qiskit/circuit/quantumcircuit.py

```diff
@@ -2560,6 +2560,10 @@ def _assign_parameter(self, parameter: Parameter, value: ParameterValueType) ->
                 assignee = instr.params[param_index]
                 # Normal ParameterExpression.
                 if isinstance(assignee, ParameterExpression):
+                    if isinstance(value, QuantumCircuit):
+                        if parameter in value.parameters:
+                            value.assign_parameters(parameter, value)
+                        continue
```

Comment:
This doesn't feel right - in what situation would we be trying to bind an expression to a `QuantumCircuit` value?

Reply:
Not sure why I put this. Removed.